main.tf

```diff
@@ -118,8 +117,7 @@ resource "azurerm_application_gateway" "ag" {
   dynamic "http_listener" {
     for_each = [for app in local.gateways[count.index].app_configuration : {
       name                 = "${app.product}-${app.component}"
-      host_name            = "${app.product}-${app.component}-${var.env}.${local.gateways[count.index].gateway_configuration.host_name_suffix}"
-      ssl_host_name        = "${app.product}-${app.component}.${local.gateways[count.index].gateway_configuration.ssl_host_name_suffix}"
+      host_name            = "${app.product}-${app.component}.${contains(keys(app), "host_name_suffix_override") ? app.host_name_suffix_override : local.gateways[count.index].gateway_configuration.host_name_suffix}"
```

Comment:
[lookup](https://www.terraform.io/docs/language/functions/lookup.html) is simpler, (used elsewhere in the file)

why do you need to override the suffix though?

Reply:
Ok, will use lookup. 

I assumed everything would be {env}.platform.hmcts.net going forward, so I made that default. Some of the listeners on the cft backend app gateway are using service.core-compute-{env}.internal, so to make it work with those you just need to set the override.

Just putting the product and component names in the backend yaml file will create a listener using {product}-{component}.{env}.platform.hmcts.net on port 80. But that's only because I assumed this should be the default, should it be different?

Reply:
will look through this with you later, don't fully understand the issue.